Jenkinsfile-utils

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env groovy
+
+/**
+ * noisePageBuild will create a build directory and compile and build the
+ * noisepage binary in that directory. The options passed into the method
+ * determine the compilation and build options. Refer to the defaultArgs
+ * for the different options that can be passed in.
+ */
+void noisePageBuild(Map args = [:]) {
+    Map defaultArgs = [
+        useCache: true,
+        buildType: 'Debug',
+        os: 'ubuntu',
+        isBuildTests: true,
+        useASAN: false,
+        isBuildBenchmarks: false,
+        isCodeCoverage: false,
+        isJumboTest: false,
+        isRecordTime: false
+    ]
+    Map config = defaultArgs << args
+    String compileCmd = generateCompileCmd(config)
+    String buildCmd = generateBuildCmd(config)
+    String buildScript = generateBuildScript(compileCmd, buildCmd, config.isRecordTime)
+
+    sh script:buildScript, label: 'Build'
+}
+
+/**
+ * generateCompileCmd creates the cmake command string. It is based on the
+ * config passed into the function. The config arguments are the same as the
+ * defaultArgs in noisePageBuild.
+ */
+String generateCompileCmd(Map config = [:]) {
```

Comment:
Nit: it seems like you want to always generate a complete command, rather than relying on the CMake noisepage defaults. Otherwise someone can change those and not know of this function, then mystery things happen in CI.

Reply:
So to be explicit, you would have every `-DNOISEPAGE_BLA_BLA` and explicitly `ON/OFF` them.

Reply:
That makes sense. That would also simplify the case you mentioned in your overall comment cause developers would no longer have to check `CMakeLists.txt`